VisualRust/VisualRustSmartIndent.cs

```diff
@@ -40,43 +43,66 @@ internal VisualRustSmartIndent(ITextView textView, IEditorOperations operations)
             ed = operations;
         }
 
-        // Check if the last non-whitespace token is left brace
-        private static bool EndsWidthLBrace(List<IToken> tokens)
+        int? ISmartIndent.GetDesiredIndentation(ITextSnapshotLine currentSnapshotLine)
         {
-            for(int i = tokens.Count - 1; i >= 0; i--)
-            {
-                if (tokens[i].Type == RustLexer.RustLexer.WHITESPACE)
-                    continue;
-                else if (tokens[i].Type == RustLexer.RustLexer.LBRACE)
-                    return true;
-                else
-                    return false;
-            }
-            throw new InvalidOperationException();
-        }
+            var textView = _textView;
+            var textSnapshot = textView.TextSnapshot;
+            var caret = textView.Caret;
+            var caretPosition = caret.Position.BufferPosition.Position;            
+            var caretLine = textSnapshot.GetLineFromPosition(caretPosition);
```

Comment:
No need to do this line-by-line. Instead you should call `textSnapshot.GetText(0, caretPosition)` here, pass the result to lexer (`Utils.LexString(line.GetText())`), and proceed as you currently do.
This will solve multi-line comments issue.